channels.go

```diff
@@ -0,0 +1,105 @@
+package mackerel
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// Channel represents a Mackerel notification channel.
+// ref. https://mackerel.io/api-docs/entry/channels
+type Channel struct {
+	ID string `json:"id"`
+	ChannelWithoutID
+}
+
+// ChannelWithoutID represents a Mackerel notification channel without the ID.
+type ChannelWithoutID struct {
```

Comment:
I understand this type is used for *CreateChannel*. But the type named XxxWithoutID is introduced for the first time at this p-r.

So I suggest to add `omitempty` tag to *Channel.ID*, or rename *ChannelWithoutID* to *CreateChannelParams*, like *CreateDashboard* or *CreateHost*.

Reply:
Aha, I see.
I feel like using `omitempty` that way is not accurate so I think I will change it to `CreateChannelParams`